cms/cms-field-type.md

```diff
@@ -0,0 +1,524 @@
+---
+title: CMS field types
+---
+
+[[toc]]
+
+## Introduction
+
+When you configure `fields` of a CMS record, you need to define the type of each field.
+
+```yml
+records:
+  blogpost:
+    record_type: blogpost
+      list:
+        fields:
+          - name: content
+            type: String # CMS field types
+            label: Content
+```
+
+### Field type =/= Data type
+
+Field type in Skygear CMS refers to the UI presentation of the field, not its data type in the database.
+
+For example, suppose there are two roles in your app: student and teacher. In the user table, there is a column named 'role' the stores the role of the user. 'role' in the database has a data type of **String**, however, in the CMS, you can set the its field type to **Dropdown** to limit the input to student or teacher.
+
+```yml
+records:
+  user:
+   record_type: user
+     list:
+       fields:
+          - name: role
+            type: Dropdown # CMS field types
+            label: Role
+            options:
+              - label: Teacher
+                value: teacher
+              - label: Student
+                value: student
+```
+
+### You can use different field type for the same field in different views
+
+A CMS record has 4 views: list, show, edit and new. It is likely that the 4 views have similar fields. However, considering the UI presentation of the 4 views are different, you may want to use different field type in different view. 
+
+Suppose you want to display the content of a blogpost as String, however, when users edit or create a new blogpost, you want to display the content of the blogpost as WYSIWYG so that users could add HTML markups to the content.
+
+``` yml
+records:
+  blogpost:
+    record_type: blogpost
+      list:
+        fields:
+          - name: content
+            type: String
+            label: Content
+      show:
+        fields:
+          - name: content
+            type: String
+            label: Content      
+      edit:
+        fields:
+          - name: content
+            type: WYSIWYG
```

Comment:
I think WYSIWYG may not be a good example, because the raw HTML would be displayed in `type: String`.

Reply:
Sorry I don't get it. Do you mean the list & show view will display the raw HTML of the field when `type: WYSIWYG` is used in the edit & new view, despite that `type: String` is used in the list & show view?

Reply:
Exactly.